cli/responseemitter.go

```diff
@@ -100,24 +67,39 @@ func (re *responseEmitter) isClosed() bool {
 }
 
 func (re *responseEmitter) Close() error {
+	return re.CloseWithError(nil)
+}
+
+func (re *responseEmitter) CloseWithError(err error) error {
```

Comment:
IMO, it's easier to read this when _all_ the close logic is in the same function.